packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,200 @@
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { LitAbility, LitActionResource } from "@lit-protocol/auth-helpers";
+import {
+  type AuthCallbackParams,
+  type AuthSig,
+  type SessionSigsMap,
+} from "@lit-protocol/types";
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { generateSessionKeyPair } from "@lit-protocol/crypto";
+
+import {
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import type { TypedDataDomain } from "viem";
+
+import {
+  type LitAuthMethod,
+  type LitSessionSigsMap,
+  type LitAccountAuthenticatorParams,
+  type LITAuthenticateProps,
+} from "./types.js";
+
+const SIGNER_TYPE: string = "lit";
+
+/**
+ * Implementation of `SmartAccountAuthenticator` for lit protocol
+ * This class requies:
+ * `@lit-protocol/lit-node-client`
+ * `@lit-protocol/pkp-ethers`
+ */
+export class LitSigner<C extends LitAuthMethod | LitSessionSigsMap>
+  implements
+    SmartAccountAuthenticator<
+      LITAuthenticateProps<C>,
+      LitSessionSigsMap,
+      PKPEthersWallet | undefined
```

Comment:
nit: for readability, can you make this a type in the corresponding types.ts file? feel free to push back.